### PR TITLE
Calculator

### PR DESCRIPTION
**Pull request by ivlytvynets**

### CODE REVIEW

**Review by vlykhvar — APPROVED** [review]

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    private static final double MULTIPLY_A_B = 5 * 17;
+    private static final double DIVIDE_A_B = (double) 5 / 17;
+    private static final double DIVIDE_B_A = (double) 17 / 5;
+    private static final double RAISE_A_TO_B = Math.pow(5, 17);
+    private static final double RAISE_B_TO_A = Math.pow(17, 5);
+    private static final int ZERO = 0;
+    private Calculator calculator = new Calculator();
+
+    @Test
+    void add_Ok() {
```
Comment (dmytro-hryhoruk) [inline src/test/java/core/basesyntax/CalculatorTest.java:25]:
```suggestion
    void addition_Ok() {
```

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+    @Test
+    void add_NotOk() {
+        assertThrows(InvalidValuesException.class, () -> {
+            calculator.calculate(Integer.MAX_VALUE, 1, '+');
+        });
+    }
+
+    @Test
+    void subtract_Ok() {
```
Comment (dmytro-hryhoruk) [inline src/test/java/core/basesyntax/CalculatorTest.java:39]:
```suggestion
    void subtraction_Ok() {
```
Let's add a full word to others too

**Review by dmytro-hryhoruk — CHANGES_REQUESTED** [review]
Let's add more testcases with negative numbers(don't forget about rasing to negative power), and make sure that constants in tests has informative naming:
DEFAULT_A -> POSITIVE_A etc.

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.exception.DivisionByZeroException;
+import core.basesyntax.exception.IllegalOperatorException;
+import core.basesyntax.exception.InvalidValuesException;
+import org.junit.jupiter.api.Test;
+
+class CalculatorTest {
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:11]: add access modifiers to class and all its methods

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    private static final int POSITIVE_B = 17;
+    private static final double ADD_A_B = 22;
+    private static final double SUBTRACT_A_B = -12;
+    private static final double MULTIPLY_A_B = 5 * 17;
+    private static final double DIVIDE_A_B = (double) 5 / 17;
+    private static final double DIVIDE_B_A = (double) 17 / 5;
+    private static final double RAISE_A_TO_B = Math.pow(5, 17);
+    private static final double RAISE_B_TO_A = Math.pow(17, 5);
+    private static final int ZERO = 0;
+    private Calculator calculator = new Calculator();
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:22]: let’s create a Calculator instance in a method that is run before all tests

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+    @Test
+    void addition_Ok() {
+        double expected = ADD_A_B;
+        double actual = calculator.calculate(POSITIVE_A, POSITIVE_B, '+');
+        assertEquals(expected, actual);
+        expected = -10;
+        actual = calculator.calculate(-2.8, -7.2, '+');
+        assertEquals(expected, actual);
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:31]: add test case for mixed positive and negative values
and for 0

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    @Test
+    void subtraction_Ok() {
+        double expected = SUBTRACT_A_B;
+        double actual = calculator.calculate(POSITIVE_A, POSITIVE_B, '-');
+        assertEquals(expected, actual);
+        expected = 4.4;
+        actual = calculator.calculate(-2.8, -7.2, '-');
+        assertEquals(expected, actual);
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:48]: same

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    @Test
+    void division_Ok() {
+        double expected = DIVIDE_A_B;
+        double actual = calculator.calculate(POSITIVE_A, POSITIVE_B, '/');
+        assertEquals(expected, actual);
+        expected = (double) -7 / 9;
+        actual = calculator.calculate(-7, 9, '/');
+        assertEquals(expected, actual);
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:65]: add test cases for 2 negative values and edge Integer values

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    @Test
+    void multiplication_Ok() {
+        double expected = MULTIPLY_A_B;
+        double actual = calculator.calculate(POSITIVE_A, POSITIVE_B, '*');
+        assertEquals(expected, actual);
+        expected = -8;
+        actual = calculator.calculate(-2, 4, '*');
+        assertEquals(expected, actual);
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:82]: same, and check for 0 as well

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+    @Test
+    void raisingToPower_Ok() {
+        double expected = RAISE_A_TO_B;
+        double actual = calculator.calculate(POSITIVE_A, POSITIVE_B, '^');
+        assertEquals(expected, actual);
+        expected = RAISE_B_TO_A;
+        actual = calculator.calculate(POSITIVE_B, POSITIVE_A, '^');
+        assertEquals(expected, actual);
+        expected = 0.5;
+        actual = calculator.calculate(2, -1, '^');
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:94]: add test case for raising to zero power

and for raising 0 to some power

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]
